openmmtools/mcmc.py

```diff
@@ -645,18 +686,16 @@ def apply(self, thermodynamic_state, sampler_state):
         move_name = self.__class__.__name__  # shortcut
         timer = Timer()
 
-        # Check if we have to use the global cache.
-        if self.context_cache is None:
-            context_cache = cache.global_context_cache
-        else:
-            context_cache = self.context_cache
-
         # Create integrator.
         integrator = self._get_integrator(thermodynamic_state)
 
+        # Get context cache
+        local_context_cache = self._get_context_cache(context_cache)
+
         # Create context.
         timer.start("{}: Context request".format(move_name))
-        context, integrator = context_cache.get_context(thermodynamic_state, integrator)
+        # TODO: Is this still needed now that we are specifying the context?
```

Comment:
Is this a future TODO or did you figure it out?

Reply:
It's still to be figured out, it just seems a bit redundant to specify a context cache and then use this `get_context`, but we have you to find out if there's a better way to accomplish this. Maybe for a future quick enhancement?